packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -45,6 +45,11 @@ type DependencyData<TSplitCondition> = $ReadOnly<{
   // If left unspecified, then the dependency is unconditionally split.
   splitCondition?: TSplitCondition,
   locs: Array<BabelSourceLocation>,
+
+  /** Should search recursively through folders. */
+  recursive?: boolean,
+  /** A filter to run against files that match the directory */
+  filter?: string,
```

Comment:
I wonder if this should be a RegExp instance (but we probably `JSON.stringify` this data in a few places, which would be problematic)